### PR TITLE
[WIP] Bad log message fix and refactor 

### PR DESCRIPTION
**Pull request by shri-acha**
Replaced the log messages for now, have to work on removing the drop count as will be discussed in #6983 .

### CODE REVIEW

File: sdk/log/record.go [hunk 1]
```diff
@@ -97,12 +101,10 @@ type Record struct {
 }
 
 func (r *Record) addDropped(n int) {
-	logAttrDropped()
 	r.dropped += n
```
Comment (pellared) [inline sdk/log/record.go:107]: would it not be easier to keep `logAttrDropped()` here just add it inside an `if`?

```go
func (r *Record) addDropped(n int) {
	r.dropped += n
	if n > 0 {
		logAttrDropped()
	}
```

File: sdk/log/record.go [hunk 2]
```diff
@@ -97,12 +101,10 @@ type Record struct {
...
 }
 
 func (r *Record) setDropped(n int) {
-	logAttrDropped()
 	r.dropped = n
 }
```
Comment (pellared) [inline sdk/log/record.go]: would it not be easier to keep `logAttrDropped()` here just add it inside an `if`?

```go
func (r *Record) setDropped(n int) {
	r.dropped = n
	if n > 0 {
		logAttrDropped()
	}
```

File: sdk/log/record.go [hunk 3]
```diff
@@ -192,11 +194,15 @@ func (r *Record) AddAttributes(attrs ...log.KeyValue) {
 	if n == 0 {
 		// Avoid the more complex duplicate map lookups below.
 		var drop int
-		attrs, drop = dedup(attrs)
-		r.setDropped(drop)
```
Comment (pellared) [inline sdk/log/record.go]: Do I see correctly that there was a bug and `logAttrDropped` was called even if nothing was actually dropped (and you fixed it as well)?

**Review by pellared — COMMENTED** [review]
Overall looks like the changes are in the good direction.

File: sdk/log/record.go (hunk 3 above)
Comment (shri-acha) [reply sdk/log/record.go]: i think so, it was to isolate the logAttrDropped and the actual setDropped and addDropped,

File: sdk/log/record.go (hunk 2 above)
Comment (shri-acha) [reply sdk/log/record.go]: I mean, either one could work.
Though, I think yours is better. I'll just quickly replace that then.

**Comment by shri-acha** [comment]
I've made the changes @pellared . I'm still unsure about how I have to deal with the tests.

**Comment by pellared** [comment]
> I've made the changes @pellared .

Can you please update the branch and resolve conficts?

>  I'm still unsure about how I have to deal with the tests.

Update/fix them if they do not test the expected behavior. Also add any missing ones (e.g. for the new log message). It would be also good to add a test for regression that the log messages are not emitted when everything is fine (no deduplication and none limit has been exceeded).
